File: Guanabara - PHP/exercicios-PHPOO/Aula12-Polimorfismo/Animal.php
<?php

abstract class Animal {
    //Atributos
    protected $peso;
    protected $idade;
    protected $membros;
    
    //Métodos Abstratos (para ser implementados nas classes filhas)
    abstract function locomover();

    abstract function alimentar();

    abstract function emitirSom();

    //Métodos Acessores
    public function getPeso() {
        return $this->peso;
    }
    public function getIdade() {
        return $this->idade;
    }
    public function getMembros() {
        return $this->membros;
    }
    public function setPeso($peso): void {
        $this->peso = $peso;
    }
    public function setIdade($idade): void {
        $this->idade = $idade;
    }
    public function setMembros($membros): void {
        $this->membros = $membros;
    }

}

File: Java/Meus exercícios POO/Aula12.Polimorfismo/src/main/java/polimorfismo/Animal.java
package polimorfismo;
public abstract  class Animal { //classe abstrata só pode servir como projenitora
    //Atributos
    protected float peso;
    protected int idade;
    protected int membros;
    
    //Métodos Abstratos (para Polimorfismo)
    public abstract void locomover();
    public abstract void alimentar();
    public abstract void emitirSom ();
    
    //Métodos Acessores
    public float getPeso() {
        return peso;
    }

    public void setPeso(float peso) {
        this.peso = peso;
    }

    public int getIdade() {
        return idade;
    }

    public void setIdade(int idade) {
        this.idade = idade;
    }

    public int getMembros() {
        return membros;
    }

    public void setMembros(int membros) {
        this.membros = membros;
    }
}
